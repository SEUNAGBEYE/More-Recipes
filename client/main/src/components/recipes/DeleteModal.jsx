import React from 'react';
import PropTypes from 'prop-types';

const DeleteModal = (props) => (
  <div>
    <div className="modal fade" id={`deleteModal${props.id}`}
      tabIndex="-1" role="dialog" aria-labelledby="modalLabel"
      aria-hidden="true"
    >
      <div className="modal-dialog" role="document">
        <div className="modal-content">
          <div className="modal-header">
            <button type="button" className="close" data-dismiss="modal"
              aria-label="Close" style={{ color: 'orange' }}
            >
              <span aria-hidden="true">&times;</span>
            </button>
            <h4 className="modal-title" id="modalLabel"
              style={{ color: 'orange' }}
            >
            Delete
            </h4>
          </div>

          <div className="modal-body">
            <p style={{ fontSize: 16 }}>
              <strong>Do you want to delete this recipe?</strong>
            </p>
          </div>

          <div className="modal-footer">
            <button className="btn btn-secondary auth-button"
              data-dismiss = "modal" onClick={props.onDelete}
              id={props.id}>Yes
            </button>
            <button type="button" className="btn btn-secondary auth-button"
              data-dismiss="modal"
            >
                  No
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
);

const propTypes = {
  id: PropTypes.number.isRequired,
  onDelete: PropTypes.func.isRequired,
};

DeleteModal.propTypes = propTypes;

export default DeleteModal;
