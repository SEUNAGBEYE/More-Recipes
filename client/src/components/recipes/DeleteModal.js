import React, {Component} from 'react';

export default class DeleteModal extends Component {
	constructor(props){
		super(props)
		this.onClick = this.onClick.bind(this)
	}

	onClick(e){
		console.log(e.target.id)
		this.props.onDelete(this.props.id)
	}

	render(){
		return (
			<div>
					<div className="modal fade" id={`deleteModal${this.props.id}`} tabIndex="-1" role="dialog" aria-labelledby="modalLabel" aria-hidden="true">
						<div className="modal-dialog" role="document">
							<div className="modal-content">
								<div className="modal-header">
									<button type="button" className="close" data-dismiss="modal" aria-label="Close" style={{color: 'orange'}}>
										<span aria-hidden="true">&times;</span>
									</button>
									<h4 className="modal-title" id="modalLabel" style={{color:'orange'}}>Delete</h4>
								</div>

								<div className='modal-body'>
									<p style={{fontSize: 16}}><strong>Do you want to delete this recipe?</strong></p>
								</div>

								<div className="modal-footer">
									<button className="btn btn-secondary auth-button" data-dismiss = "modal" onClick={this.onClick} id={this.props.id}>Yes</button>
									<button type="button" className="btn btn-secondary auth-button" data-dismiss="modal">
										No
									</button>
								</div>
							</div>
						</div>
					</div>
			</div>
		)
 }
}