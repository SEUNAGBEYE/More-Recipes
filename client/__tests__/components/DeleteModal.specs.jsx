
import React from 'react';
import { shallow } from 'enzyme';
import DeleteModal from '../../main/src/components/recipes/DeleteModal';

const props = {
  onDelete: jest.fn(),
  id: 1
};

const event = {
  preventDefault: jest.fn(),
  target: {
    search: {
      value: '',
    },
    id: 1
  }
}

describe('# CategoryButton', () => {
  const wrapper = shallow(
      <DeleteModal {...props} />
  );
  it('should render successfully', (done) => {
    expect(wrapper).toBeDefined();
    expect(wrapper.getElement().type).toBe('div');
    expect(wrapper).toMatchSnapshot()
    done()
  });

});
