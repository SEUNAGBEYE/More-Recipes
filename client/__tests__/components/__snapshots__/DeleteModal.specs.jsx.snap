// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`# CategoryButton should render successfully 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <DeleteModal
    id={1}
    onDelete={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        aria-hidden="true"
        aria-labelledby="modalLabel"
        className="modal fade"
        id="deleteModal1"
        role="dialog"
        tabIndex="-1"
      >
        <div
          className="modal-dialog"
          role="document"
        >
          <div
            className="modal-content"
          >
            <div
              className="modal-header"
            >
              <button
                aria-label="Close"
                className="close"
                data-dismiss="modal"
                style={
                  Object {
                    "color": "orange",
                  }
                }
                type="button"
              >
                <span
                  aria-hidden="true"
                >
                  ×
                </span>
              </button>
              <h4
                className="modal-title"
                id="modalLabel"
                style={
                  Object {
                    "color": "orange",
                  }
                }
              >
                Delete
              </h4>
            </div>
            <div
              className="modal-body"
            >
              <p
                style={
                  Object {
                    "fontSize": 16,
                  }
                }
              >
                <strong>
                  Do you want to delete this recipe?
                </strong>
              </p>
            </div>
            <div
              className="modal-footer"
            >
              <button
                className="btn btn-secondary auth-button"
                data-dismiss="modal"
                id={1}
                onClick={[MockFunction]}
              >
                Yes
              </button>
              <button
                className="btn btn-secondary auth-button"
                data-dismiss="modal"
                type="button"
              >
                No
              </button>
            </div>
          </div>
        </div>
      </div>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "aria-hidden": "true",
        "aria-labelledby": "modalLabel",
        "children": <div
          className="modal-dialog"
          role="document"
        >
          <div
            className="modal-content"
          >
            <div
              className="modal-header"
            >
              <button
                aria-label="Close"
                className="close"
                data-dismiss="modal"
                style={
                  Object {
                    "color": "orange",
                  }
                }
                type="button"
              >
                <span
                  aria-hidden="true"
                >
                  ×
                </span>
              </button>
              <h4
                className="modal-title"
                id="modalLabel"
                style={
                  Object {
                    "color": "orange",
                  }
                }
              >
                Delete
              </h4>
            </div>
            <div
              className="modal-body"
            >
              <p
                style={
                  Object {
                    "fontSize": 16,
                  }
                }
              >
                <strong>
                  Do you want to delete this recipe?
                </strong>
              </p>
            </div>
            <div
              className="modal-footer"
            >
              <button
                className="btn btn-secondary auth-button"
                data-dismiss="modal"
                id={1}
                onClick={[MockFunction]}
              >
                Yes
              </button>
              <button
                className="btn btn-secondary auth-button"
                data-dismiss="modal"
                type="button"
              >
                No
              </button>
            </div>
          </div>
        </div>,
        "className": "modal fade",
        "id": "deleteModal1",
        "role": "dialog",
        "tabIndex": "-1",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="modal-content"
          >
            <div
              className="modal-header"
            >
              <button
                aria-label="Close"
                className="close"
                data-dismiss="modal"
                style={
                  Object {
                    "color": "orange",
                  }
                }
                type="button"
              >
                <span
                  aria-hidden="true"
                >
                  ×
                </span>
              </button>
              <h4
                className="modal-title"
                id="modalLabel"
                style={
                  Object {
                    "color": "orange",
                  }
                }
              >
                Delete
              </h4>
            </div>
            <div
              className="modal-body"
            >
              <p
                style={
                  Object {
                    "fontSize": 16,
                  }
                }
              >
                <strong>
                  Do you want to delete this recipe?
                </strong>
              </p>
            </div>
            <div
              className="modal-footer"
            >
              <button
                className="btn btn-secondary auth-button"
                data-dismiss="modal"
                id={1}
                onClick={[MockFunction]}
              >
                Yes
              </button>
              <button
                className="btn btn-secondary auth-button"
                data-dismiss="modal"
                type="button"
              >
                No
              </button>
            </div>
          </div>,
          "className": "modal-dialog",
          "role": "document",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="modal-header"
              >
                <button
                  aria-label="Close"
                  className="close"
                  data-dismiss="modal"
                  style={
                    Object {
                      "color": "orange",
                    }
                  }
                  type="button"
                >
                  <span
                    aria-hidden="true"
                  >
                    ×
                  </span>
                </button>
                <h4
                  className="modal-title"
                  id="modalLabel"
                  style={
                    Object {
                      "color": "orange",
                    }
                  }
                >
                  Delete
                </h4>
              </div>,
              <div
                className="modal-body"
              >
                <p
                  style={
                    Object {
                      "fontSize": 16,
                    }
                  }
                >
                  <strong>
                    Do you want to delete this recipe?
                  </strong>
                </p>
              </div>,
              <div
                className="modal-footer"
              >
                <button
                  className="btn btn-secondary auth-button"
                  data-dismiss="modal"
                  id={1}
                  onClick={[MockFunction]}
                >
                  Yes
                </button>
                <button
                  className="btn btn-secondary auth-button"
                  data-dismiss="modal"
                  type="button"
                >
                  No
                </button>
              </div>,
            ],
            "className": "modal-content",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <button
                    aria-label="Close"
                    className="close"
                    data-dismiss="modal"
                    style={
                      Object {
                        "color": "orange",
                      }
                    }
                    type="button"
                  >
                    <span
                      aria-hidden="true"
                    >
                      ×
                    </span>
                  </button>,
                  <h4
                    className="modal-title"
                    id="modalLabel"
                    style={
                      Object {
                        "color": "orange",
                      }
                    }
                  >
                    Delete
                  </h4>,
                ],
                "className": "modal-header",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "aria-label": "Close",
                    "children": <span
                      aria-hidden="true"
                    >
                      ×
                    </span>,
                    "className": "close",
                    "data-dismiss": "modal",
                    "style": Object {
                      "color": "orange",
                    },
                    "type": "button",
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "aria-hidden": "true",
                      "children": "×",
                    },
                    "ref": null,
                    "rendered": "×",
                    "type": "span",
                  },
                  "type": "button",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Delete",
                    "className": "modal-title",
                    "id": "modalLabel",
                    "style": Object {
                      "color": "orange",
                    },
                  },
                  "ref": null,
                  "rendered": "Delete",
                  "type": "h4",
                },
              ],
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <p
                  style={
                    Object {
                      "fontSize": 16,
                    }
                  }
                >
                  <strong>
                    Do you want to delete this recipe?
                  </strong>
                </p>,
                "className": "modal-body",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <strong>
                    Do you want to delete this recipe?
                  </strong>,
                  "style": Object {
                    "fontSize": 16,
                  },
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Do you want to delete this recipe?",
                  },
                  "ref": null,
                  "rendered": "Do you want to delete this recipe?",
                  "type": "strong",
                },
                "type": "p",
              },
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <button
                    className="btn btn-secondary auth-button"
                    data-dismiss="modal"
                    id={1}
                    onClick={[MockFunction]}
                  >
                    Yes
                  </button>,
                  <button
                    className="btn btn-secondary auth-button"
                    data-dismiss="modal"
                    type="button"
                  >
                    No
                  </button>,
                ],
                "className": "modal-footer",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Yes",
                    "className": "btn btn-secondary auth-button",
                    "data-dismiss": "modal",
                    "id": 1,
                    "onClick": [MockFunction],
                  },
                  "ref": null,
                  "rendered": "Yes",
                  "type": "button",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "No",
                    "className": "btn btn-secondary auth-button",
                    "data-dismiss": "modal",
                    "type": "button",
                  },
                  "ref": null,
                  "rendered": "No",
                  "type": "button",
                },
              ],
              "type": "div",
            },
          ],
          "type": "div",
        },
        "type": "div",
      },
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          aria-hidden="true"
          aria-labelledby="modalLabel"
          className="modal fade"
          id="deleteModal1"
          role="dialog"
          tabIndex="-1"
        >
          <div
            className="modal-dialog"
            role="document"
          >
            <div
              className="modal-content"
            >
              <div
                className="modal-header"
              >
                <button
                  aria-label="Close"
                  className="close"
                  data-dismiss="modal"
                  style={
                    Object {
                      "color": "orange",
                    }
                  }
                  type="button"
                >
                  <span
                    aria-hidden="true"
                  >
                    ×
                  </span>
                </button>
                <h4
                  className="modal-title"
                  id="modalLabel"
                  style={
                    Object {
                      "color": "orange",
                    }
                  }
                >
                  Delete
                </h4>
              </div>
              <div
                className="modal-body"
              >
                <p
                  style={
                    Object {
                      "fontSize": 16,
                    }
                  }
                >
                  <strong>
                    Do you want to delete this recipe?
                  </strong>
                </p>
              </div>
              <div
                className="modal-footer"
              >
                <button
                  className="btn btn-secondary auth-button"
                  data-dismiss="modal"
                  id={1}
                  onClick={[MockFunction]}
                >
                  Yes
                </button>
                <button
                  className="btn btn-secondary auth-button"
                  data-dismiss="modal"
                  type="button"
                >
                  No
                </button>
              </div>
            </div>
          </div>
        </div>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "aria-hidden": "true",
          "aria-labelledby": "modalLabel",
          "children": <div
            className="modal-dialog"
            role="document"
          >
            <div
              className="modal-content"
            >
              <div
                className="modal-header"
              >
                <button
                  aria-label="Close"
                  className="close"
                  data-dismiss="modal"
                  style={
                    Object {
                      "color": "orange",
                    }
                  }
                  type="button"
                >
                  <span
                    aria-hidden="true"
                  >
                    ×
                  </span>
                </button>
                <h4
                  className="modal-title"
                  id="modalLabel"
                  style={
                    Object {
                      "color": "orange",
                    }
                  }
                >
                  Delete
                </h4>
              </div>
              <div
                className="modal-body"
              >
                <p
                  style={
                    Object {
                      "fontSize": 16,
                    }
                  }
                >
                  <strong>
                    Do you want to delete this recipe?
                  </strong>
                </p>
              </div>
              <div
                className="modal-footer"
              >
                <button
                  className="btn btn-secondary auth-button"
                  data-dismiss="modal"
                  id={1}
                  onClick={[MockFunction]}
                >
                  Yes
                </button>
                <button
                  className="btn btn-secondary auth-button"
                  data-dismiss="modal"
                  type="button"
                >
                  No
                </button>
              </div>
            </div>
          </div>,
          "className": "modal fade",
          "id": "deleteModal1",
          "role": "dialog",
          "tabIndex": "-1",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <div
              className="modal-content"
            >
              <div
                className="modal-header"
              >
                <button
                  aria-label="Close"
                  className="close"
                  data-dismiss="modal"
                  style={
                    Object {
                      "color": "orange",
                    }
                  }
                  type="button"
                >
                  <span
                    aria-hidden="true"
                  >
                    ×
                  </span>
                </button>
                <h4
                  className="modal-title"
                  id="modalLabel"
                  style={
                    Object {
                      "color": "orange",
                    }
                  }
                >
                  Delete
                </h4>
              </div>
              <div
                className="modal-body"
              >
                <p
                  style={
                    Object {
                      "fontSize": 16,
                    }
                  }
                >
                  <strong>
                    Do you want to delete this recipe?
                  </strong>
                </p>
              </div>
              <div
                className="modal-footer"
              >
                <button
                  className="btn btn-secondary auth-button"
                  data-dismiss="modal"
                  id={1}
                  onClick={[MockFunction]}
                >
                  Yes
                </button>
                <button
                  className="btn btn-secondary auth-button"
                  data-dismiss="modal"
                  type="button"
                >
                  No
                </button>
              </div>
            </div>,
            "className": "modal-dialog",
            "role": "document",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <div
                  className="modal-header"
                >
                  <button
                    aria-label="Close"
                    className="close"
                    data-dismiss="modal"
                    style={
                      Object {
                        "color": "orange",
                      }
                    }
                    type="button"
                  >
                    <span
                      aria-hidden="true"
                    >
                      ×
                    </span>
                  </button>
                  <h4
                    className="modal-title"
                    id="modalLabel"
                    style={
                      Object {
                        "color": "orange",
                      }
                    }
                  >
                    Delete
                  </h4>
                </div>,
                <div
                  className="modal-body"
                >
                  <p
                    style={
                      Object {
                        "fontSize": 16,
                      }
                    }
                  >
                    <strong>
                      Do you want to delete this recipe?
                    </strong>
                  </p>
                </div>,
                <div
                  className="modal-footer"
                >
                  <button
                    className="btn btn-secondary auth-button"
                    data-dismiss="modal"
                    id={1}
                    onClick={[MockFunction]}
                  >
                    Yes
                  </button>
                  <button
                    className="btn btn-secondary auth-button"
                    data-dismiss="modal"
                    type="button"
                  >
                    No
                  </button>
                </div>,
              ],
              "className": "modal-content",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <button
                      aria-label="Close"
                      className="close"
                      data-dismiss="modal"
                      style={
                        Object {
                          "color": "orange",
                        }
                      }
                      type="button"
                    >
                      <span
                        aria-hidden="true"
                      >
                        ×
                      </span>
                    </button>,
                    <h4
                      className="modal-title"
                      id="modalLabel"
                      style={
                        Object {
                          "color": "orange",
                        }
                      }
                    >
                      Delete
                    </h4>,
                  ],
                  "className": "modal-header",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "aria-label": "Close",
                      "children": <span
                        aria-hidden="true"
                      >
                        ×
                      </span>,
                      "className": "close",
                      "data-dismiss": "modal",
                      "style": Object {
                        "color": "orange",
                      },
                      "type": "button",
                    },
                    "ref": null,
                    "rendered": Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "aria-hidden": "true",
                        "children": "×",
                      },
                      "ref": null,
                      "rendered": "×",
                      "type": "span",
                    },
                    "type": "button",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Delete",
                      "className": "modal-title",
                      "id": "modalLabel",
                      "style": Object {
                        "color": "orange",
                      },
                    },
                    "ref": null,
                    "rendered": "Delete",
                    "type": "h4",
                  },
                ],
                "type": "div",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <p
                    style={
                      Object {
                        "fontSize": 16,
                      }
                    }
                  >
                    <strong>
                      Do you want to delete this recipe?
                    </strong>
                  </p>,
                  "className": "modal-body",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": <strong>
                      Do you want to delete this recipe?
                    </strong>,
                    "style": Object {
                      "fontSize": 16,
                    },
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Do you want to delete this recipe?",
                    },
                    "ref": null,
                    "rendered": "Do you want to delete this recipe?",
                    "type": "strong",
                  },
                  "type": "p",
                },
                "type": "div",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <button
                      className="btn btn-secondary auth-button"
                      data-dismiss="modal"
                      id={1}
                      onClick={[MockFunction]}
                    >
                      Yes
                    </button>,
                    <button
                      className="btn btn-secondary auth-button"
                      data-dismiss="modal"
                      type="button"
                    >
                      No
                    </button>,
                  ],
                  "className": "modal-footer",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Yes",
                      "className": "btn btn-secondary auth-button",
                      "data-dismiss": "modal",
                      "id": 1,
                      "onClick": [MockFunction],
                    },
                    "ref": null,
                    "rendered": "Yes",
                    "type": "button",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "No",
                      "className": "btn btn-secondary auth-button",
                      "data-dismiss": "modal",
                      "type": "button",
                    },
                    "ref": null,
                    "rendered": "No",
                    "type": "button",
                  },
                ],
                "type": "div",
              },
            ],
            "type": "div",
          },
          "type": "div",
        },
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
